### PR TITLE
feat: multi fee implementation

### PR DESCRIPTION
**Pull request by ozwaldorf**
## Why?

need to handle multiple fees; protocol fee, collection fee, and in the future a frontend fee

## How?

- added fee vec tuple to Listing datatype, for frontend to be able to use
- added protocol_fee to init args
- moved `owner_fee_percentage` to collection_fee in collection data
- added getCollections, returns vec of collection info (includes collection_fee)
- added getProtocolFee, returns nat fee e2 

## Tickets?

- [Ticket 1](the-ticket-url-here)
- [Ticket 2](the-ticket-url-here)
- [Ticket 3](the-ticket-url-here)

## Contribution checklist?

- [ ] The commit messages are detailed
- [ ] It does not break existing features (unless required)
- [ ] I have performed a self-review of my own code
- [ ] Documentation has been updated to reflect the changes
- [ ] Tests have been added or updated to reflect the changes
- [ ] All code formatting pass
- [ ] All lints pass
- [ ] All tests pass

## Security checklist?

- [ ] Injection has been prevented (parameterized queries, no eval or system calls)
- [ ] Sensitive data has been identified and is being protected properly

## Demo?

Optionally, provide any screenshot, gif or small video.

### CODE REVIEW

**Comment by ozwaldorf** [comment]
rebased to develop without squash, apologies for the commit message confusion

File: marketplace/src/main.rs [hunk 1]
```diff
@@ -212,6 +266,21 @@ fn add_collection(
     Ok(())
 }
 
+#[update(name = "setProtocolFee")]
+#[candid_method(update, rename = "setProtocolFee")]
+fn set_protocol_fee(fee: Nat) -> MPApiResult {
+    let init_data = init_data().clone();
```
Comment (punkbit) [inline marketplace/src/main.rs:272]: We need to remove the todo in https://github.com/Psychedelic/nft-marketplace/blob/develop/marketplace/src/utils.rs#L24 find init_data

**Review by punkbit — APPROVED** [review]
@ozwaldorf a commit before, had `.seller` which replaces `.payment_address`, but we see that it's also happening here.

As discussed earlier, for any interface changes, we should have the related changes in FE, to avoid confusion. So, where required, make those changes in FE along with the integration of this  PR, to avoid blockers. Thanks!

**Comment by ozwaldorf** [comment]
payment_address changes is already merged into here and also merged into FE with my PR. Was all at once with the token_owner addition

**Pull request merged**